WooCommerce/Classes/ViewModels/Order Details/OrderStatusListViewModel.swift

```diff
@@ -15,8 +15,10 @@ final class OrderStatusListViewModel {
     ///
     var indexOfSelectedStatus: IndexPath? {
         didSet {
-            if initialStatus != indexOfSelectedStatus {
+            if initialStatus != indexOfSelectedStatus, !isSelectionAutoConfirmed {
                 shouldEnableApplyButton = true
+            } else if initialStatus != indexOfSelectedStatus, isSelectionAutoConfirmed {
+                confirmSelectedStatus()
             } else {
                 shouldEnableApplyButton = false
             }
```

Comment:
Do you think that having early returns or a `switch` will make these lines easier to follow?

Something like:
```swift
let selectedNewStatus = initialStatus != indexOfSelectedStatus
switch (selectedNewStatus, isSelectionAutoConfirmed) {
case (true, false):
    shouldEnableApplyButton = true // New status & manual confirmation
case (true, true):
    confirmSelectedStatus() // New status & automatic confirmation
case (false, _):
    shouldEnableApplyButton = false // No new status
    }
}```

Reply:
Good suggestion to use a `switch` here; that's much clearer!